test/template-tests/testcases/deploymentTemplate/apiVersions-Should-Be-Recent.test.ps1

```diff
@@ -0,0 +1,34 @@
+﻿param(
+[Parameter(Mandatory=$true,Position=0)]
+[PSObject]
+$MainTemplateObject
+)
+
+$allApiVersions = $MainTemplateObject | 
+    Find-AzureRMTemplate -Key apiVersion -Value * -Like
+
+foreach ($av in $allApiVersions) {
+    if ($av.ApiVersion -isnot [string]) {
+        Write-Error "Api Versions must be strings" -TargetObject $av -ErrorId ApiVersion.Not.String
+        continue
+    }
+    $helperName = if ($av.type) {
+        $av.type
+    } elseif ($av.name) {
+        $av.name
+    } else {
+        ''
+    }
+
+    $apiDate = $av.ApiVersion -as [DateTime]
+    if (-not $apiDate) {
+        Write-Error "Api versions must be a fixed date: ($helperName is not)" -TargetObject $av -ErrorId ApiVersion.Not.Date
+        continue
+    }
+    
+
+    $timeSinceApi = [DateTime]::Now - $apiDate
+    if ($timeSinceApi.TotalDays -gt 365) {

```

Comment:
need to chat about this one - we'll need to build an "index" of apiVerstion that are allowed